### PR TITLE
Updating the design checklist in component checklist

### PR DESCRIPTION
**Pull request by heatherlarsen**
### :pushpin: Summary

If merged, this PR updates the New Component Checklist to include changes to the design workflow since the website was launched. 

**Preview link:** https://github.com/hashicorp/design-system/blob/hl-component-checklist/packages/components/NEW-COMPONENT-CHECKLIST.md

### :link: External links

Jira ticket: [HDS-1794](https://hashicorp.atlassian.net/browse/HDS-1794)

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A11y tests have been run locally (`yarn test:a11y --filter="COMPONENT-NAME"`)
- [ ] If documenting a new component, an acceptance test that includes the `a11yAudit` has been added
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-1794]: https://hashicorp.atlassian.net/browse/HDS-1794?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by didoo — APPROVED** [review]
All good for me. 👍

File: packages/components/NEW-COMPONENT-CHECKLIST.md [hunk 1]
```diff
@@ -1,70 +1,89 @@
...
+- [ ] while the component is being tested, write the documentation
+  - artifacts should include:
+    - design guidelines
+    - specifications
+    - relevant accessibility details
+       
+### Release in Figma
+Once ambassadors have tested the component and iterations are approved, prepare for a final release of the component.
+
+- [ ] add a changelog entry to [CHANGELOG-FIGMA-COMPONENTS.md]([https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components?type=design&node-id=6790-10926&mode=design](https://github.com/hashicorp/design-system/blob/main/packages/components/CHANGELOG-FIGMA-COMPONENTS.md))
```
Comment (Dhaulagiri) [inline packages/components/NEW-COMPONENT-CHECKLIST.md:84]: should this link only to the changelog in github?

Comment (heatherlarsen) [reply packages/components/NEW-COMPONENT-CHECKLIST.md]: Fixed it!

**Review by jorytindall — APPROVED** [review]
:shipit:

**Pull request merged**